torch/distributed/fsdp/api.py

```diff
@@ -151,13 +156,34 @@ class MixedPrecision:
         not use an ``auto_wrap_policy``, then the user must take care to not
         use mixed precision for FSDP instances containing ``BatchNorm``
         modules.
+
+    .. note:: ``cast_root_forward_inputs`` is set as True and ``cast_forward_inputs`` is set
+        as False in default for every FSDP isntance. For root FSDP instance,
+        ``cast_root_forward_inputs`` takes precedence over ``cast_forward_inputs``,
+        for non-root FSDP instance, ``cast_root_forward_inputs`` is ignored.
+        This is usually sufficient for the case where every FSDP instance
+        has the same ``MixedPrecision`` configuration, and only need to cast inputs at the
+        beginning of executing the model's forward pass.
+
+    .. note:: For submodules with different ``MixedPrecision`` configurations, it is
+        recommended to use ``cast_forward_inputs`` to configure casting inputs or not
+        for its FSDP instance. For root FSDP instance, make sure its wrapped submodules
+        are first ones to be executed, e.g. for the case
+        FSDP(FSDP(model.c2, MixedPrecision(param_dtype=torch.float16, cast_forward_inputs=True)),
+        model.c1, MixedPrecision(param_dtype=torch.bfloat16, cast_forward_inputs=True)),
+        model.c1 should be the first one executed, so that its inputs could be casted
+        as expected inside the root FSDP instance.see examples in unit tests
```

Comment:
I am not sure if we want to reference the unit tests like this from public docs since (1) users should not need to dig into our unit tests to understand the note and (2) the unit test may change without us remembering to change this public note.

We do not need to change address this in this PR. I can submit a follow-up if you do not mind, or you can do it as well.